packages/opentelemetry-core/src/context/propagation/composite.ts

```diff
@@ -0,0 +1,45 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Carrier, Context, HttpTextFormat } from '@opentelemetry/api';
+
+/** Combines multiple propagators into a single propagator. */
+export class CompositePropagator implements HttpTextFormat {
+    private _propagators: HttpTextFormat[];
+    constructor(...propagators: HttpTextFormat[]) {
+        this._propagators = propagators;
+    }
+
+    inject(context: Context, carrier: Carrier) {
+        for (const propagator of this._propagators) {
+            try {
+                propagator.inject(context, carrier);
+            } catch { }
+        }
+    }
+
+    extract(context: Context, carrier: Carrier): Context {
+        return this._propagators.reduce(
+            (ctx, propagator) => {
+                try {
+                    return propagator.extract(ctx, carrier)
```

Comment:
I like this approach, unlike otel-Go where extract would run against all configured `_propagators`: which means last one is gonna win.

Reply:
This does run against all propagators. That is necessary because the first may extract span context, but the second may extract correlations.

Reply:
I can run them in reverse order so the first one in the list runs last and is the "winner" by default?

Reply:
I had a discussion with @rghetia, he has created a spec issue: https://github.com/open-telemetry/opentelemetry-specification/issues/496